src/main/kotlin/Shovel.kt

```diff
@@ -0,0 +1,76 @@
+import Config.S3AWS.bucket
+import com.amazonaws.services.s3.AmazonS3
+import com.amazonaws.services.s3.model.ObjectMetadata
+import com.amazonaws.services.s3.model.PutObjectRequest
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.async
+import kotlinx.coroutines.isActive
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import java.io.BufferedInputStream
+import java.text.SimpleDateFormat
+import java.time.Duration
+import java.util.*
+
+val log: Logger = LoggerFactory.getLogger("shovelAsync")
+
+fun shovelAsync(kafka: KafkaConsumer<ByteArray, ByteArray>, s3client: AmazonS3, pollTimeout: Duration) =
+    GlobalScope.async {
+        log.info(Config.Kafka.reportTopicSubscriptionDetails())
+        while (isActive) {
+            kafka.subscribe(setOf(Config.Kafka.deadLetterQueueTopic))
```

Comment:
What if the dead letter queue doesn't exist? Should this application even start?

Reply:
There is a call in to create all the DLQs.